instrumentation/runtime/runtime.go

```diff
@@ -117,34 +120,44 @@ func Start(opts ...Option) error {
 
 func (r *runtime) register() error {
 	startTime := time.Now()
-	if _, err := r.meter.NewInt64CounterObserver(
+	uptime, err := r.meter.AsyncInt64().UpDownCounter(
 		"runtime.uptime",
-		func(_ context.Context, result metric.Int64ObserverResult) {
-			result.Observe(time.Since(startTime).Milliseconds())
-		},
-		metric.WithUnit(unit.Milliseconds),
-		metric.WithDescription("Milliseconds since application was initialized"),
-	); err != nil {
+		instrument.WithUnit(unit.Milliseconds),
+		instrument.WithDescription("Milliseconds since application was initialized"),
+	)
+	if err != nil {
 		return err
 	}
 
-	if _, err := r.meter.NewInt64UpDownCounterObserver(
+	goroutines, err := r.meter.AsyncInt64().UpDownCounter(
 		"process.runtime.go.goroutines",
-		func(_ context.Context, result metric.Int64ObserverResult) {
-			result.Observe(int64(goruntime.NumGoroutine()))
-		},
-		metric.WithDescription("Number of goroutines that currently exist"),
-	); err != nil {
+		instrument.WithDescription("Number of goroutines that currently exist"),
+	)
+	if err != nil {
 		return err
 	}
 
-	if _, err := r.meter.NewInt64CounterObserver(
+	cgoCalls, err := r.meter.AsyncInt64().UpDownCounter(
 		"process.runtime.go.cgo.calls",
-		func(_ context.Context, result metric.Int64ObserverResult) {
-			result.Observe(goruntime.NumCgoCall())
+		instrument.WithDescription("Number of cgo calls made by the current process"),
+	)
+	if err != nil {
+		return err
+	}
+
+	err = r.meter.RegisterCallback(
```

Comment:
Should this be three separate callbacks as it was before?  I'm not sure what impact combining them has.

Reply:
I don't believe that any of the callbacks are run in separate routines.  So this would just dispense with the overhead of tracking 3 different closures and calling each one serially.